components/ActionSheet/src/MDCActionSheetController.h

```diff
@@ -0,0 +1,179 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+#ifndef MDC_SUBCLASSING_RESTRICTED
+#if defined(__has_attribute) && __has_attribute(objc_subclassing_restricted)
+#define MDC_SUBCLASSING_RESTRICTED __attribute__((objc_subclassing_restricted))
+#else
+#define MDC_SUBCLASSING_RESTRICTED
+#endif
+#endif  // #ifndef MDC_SUBCLASSING_RESTRICTED
+
+@class MDCActionSheetAction;
+
+/**
+ MDCActionSheetController displays an alert message to the user, similar to
+     UIAlertControllerStyleActionSheet.
+
+ A Material Action Sheet consist of a title, message and a list of actions.
+
+ Learn more at the [Material spec bottom
+ sheet](https://material.io/design/components/sheets-bottom.html)
+ [Material spec list](https://material.io/design/components/lists.html)
+
+ To learn more about
+ [UIAlertController](https://developer.apple.com/documentation/uikit/uialertcontroller)
+ or [UIAlertControllerSytleActionSheet](https://developer.apple.com/documentation/uikit/uialertcontrollerstyle/uialertcontrollerstyleactionsheet)
+
+ MDCActionSheetController does not support UIPopoverController, instead it will always be presented
+ in a sheet from the bottom.
+
+ ### Dependencies
+
+ MDCActionSheetController depends on BottomSheet Component.
+ */
+MDC_SUBCLASSING_RESTRICTED
+@interface MDCActionSheetController : UIViewController
```

Comment:
Did we deem [text field stuff](https://developer.apple.com/documentation/uikit/uialertcontroller/1620093-addtextfield) overly complex for a first pass? Or are we just not concerned about it for this? Now that I think about it I don't think I've ever seen a text field in an action sheet, just an alert... 🤔

Reply:
I think Action Sheet will just be a list of items.